src/TTImp/TTImp.idr

```diff
@@ -120,62 +120,67 @@ mutual
        Unique : AltType
        UniqueDefault : RawImp -> AltType
 
+  private
+  wrapMaybe : Show a => Maybe a -> String
+  wrapMaybe x@(Just _) = "(" ++ show x ++ ")"
+  wrapMaybe x@Nothing  = show x
+
```

Comment:
I think this would be better on src/Utils/String.idr in case someone might want to reuse it on other Show instances.

Reply:
Yeah I was no the fence about that. Wasn't sure if it was too specific a use to put in a Util module. I don't mind adding it though.
Can you think of a better name than wrapMaybe? wrapShowMaybe is feeling a little verbose though it's direct

Reply:
Oh now that I think about naming it might be a good use case for named implementations

`[wrapJust] Show (Maybe a)` and then when used it would be `show @{wrapJust}`

Reply:
Neat application of instances but don't think I'd want to have an extra Show instance hanging around in the codebase just for that, in the end I'm just wrapping two uses of Just :>
(modular) wrapping functions have more potential for re-use than a specific Show instance and are slightly more discoverable since people are likely to look in Utils like you did.